src/game.rs

```diff
@@ -1,220 +1,277 @@
+
+        if let Some(cell) = self
+            .cells
+            .get_mut(cell_y)
+            .and_then(|row| row.get_mut(cell_x))
+        {
+            cell.is_dead = !cell.is_dead;
+        }
+    }
+
     fn draw_cells(&self) {
         for cells in self.cells.iter() {
-            cells.iter().for_each(|cell| {
+            cells.iter().filter(|cell| !cell.is_dead).for_each(|cell| {
                 draw_rectangle(
-                    cell.pos.x,
-                    cell.pos.y,
-                    self.cell_size,
-                    self.cell_size,
-                    if cell.is_dead { BLACK } else { WHITE },
+                    cell.pos.x as f32,
+                    cell.pos.y as f32,
+                    self.cell_size as f32,
+                    self.cell_size as f32,
+                    WHITE,
                 )
             });
         }
     }
 
-    fn get_new_generation(&self) -> CellMatrix {
-        self.cells
-            .iter()
-            .enumerate()
-            .map(|(row_idx, row)| {
-                row.iter()
-                    .enumerate()
-                    .map(|(col_idx, col)| {
-                        let count = self.get_neighbors_count(row_idx, col_idx);
-                        let is_dead = match count {
-                            3 => false,
-                            2..=3 => false,
-                            _ => true,
-                        };
-                        Cell { is_dead, ..*col }
-                    })
-                    .collect()
-            })
-            .collect()
+    fn draw_grid(&self) {
+        let num_of_rows = screen_height() as u32 / self.cell_size;
+        let num_of_cols = screen_width() as u32 / self.cell_size;
+
+        for idx in 0..=num_of_rows {
+            let y = (idx * self.cell_size) as f32;
+            draw_line(0.0, y, screen_width(), y, 1.0, GRAY);
+        }
+
+        for idx in 0..=num_of_cols {
+            let x = (idx * self.cell_size) as f32;
+            draw_line(x, 0.0, x, screen_height(), 1.0, GRAY);
+        }
     }
 
-    fn get_neighbors_count(&self, row_idx: usize, col_idx: usize) -> i32 {
-        let cells: &CellMatrix = &self.cells;
-        let start_row = if row_idx == 0 { 0 } else { row_idx - 1 };
-        let end_row = if row_idx >= cells.len() - 1 {
-            cells.len()
-        } else {
-            row_idx + 2
-        };
+    fn draw_config_bar(&mut self) {
+        widgets::Window::new(100, vec2(screen_width() + 140., 0.), vec2(500., 180.))
+            .movable(true)
+            .label("Config")
+            .ui(&mut root_ui(), |ui| {
+                let range: Range<f32> = 0.0..100.0;
+                ui.slider(1, "speed in ms", range, &mut self.state.speed_in_ms);
+
+                if ui.button(
+                    vec2(0., 50.),
+                    if self.state.is_paused {
+                        "Start"
+                    } else {
+                        "Pause"
+                    },
```

Comment:
This `if` is hard to read inside this function call. It'd be better to have it outside.